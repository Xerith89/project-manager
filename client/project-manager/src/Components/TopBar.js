import React from 'react'
import PropTypes from 'prop-types'

export default function TopBar(props) {
    return (
        <div data-testid="topbar">
         <nav className="navbar navbar-expand-lg navbar-dark bg-dark p-0">
             <h3 className="text-white ml-5 mb-0">Project Manager</h3>
             <div className="collapse navbar-collapse" id="navbarSupportedContent">
                 <ul className="navbar-nav ml-auto">
                     <li className="nav-item nav-link active">
                         <h3>Welcome, {props.user}</h3>
                     </li>
                 </ul>
             </div>
         </nav>
     </div>
     )
 }


TopBar.propTypes = {
 user: PropTypes.string
};
