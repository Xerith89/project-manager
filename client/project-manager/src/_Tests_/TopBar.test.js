import TopBar from '../Components/TopBar'
import React from 'react'
import {shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});

test('TopBar component renders to the DOM', () => {
    const wrapper = shallow(<TopBar user={"Test User"}/>);
    expect(wrapper.html()).toMatchSnapshot();
});