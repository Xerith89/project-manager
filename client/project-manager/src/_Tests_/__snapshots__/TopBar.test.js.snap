// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component renders to the DOM 1`] = `"<div data-testid=\\"topbar\\"><nav class=\\"navbar navbar-expand-lg navbar-dark bg-dark p-0\\"><h3 class=\\"text-white ml-5 mb-0\\">Project Manager</h3><div class=\\"collapse navbar-collapse\\" id=\\"navbarSupportedContent\\"><ul class=\\"navbar-nav ml-auto\\"><li class=\\"nav-item nav-link active\\"><h3>Welcome, Test User</h3></li></ul></div></nav></div>"`;
